src/main/java/seedu/address/model/person/Person.java

```diff
@@ -316,6 +335,35 @@ public boolean isSameCluster(ReadOnlyPerson other) {
     }
 
     @Override
+    public String calcAccruedAmount() {
+        this.lastAccruedDate = new Date(); // update last accrued date
+        double principal = this.getDebt().toNumber();
+        int interest = Integer.parseInt(this.getInterest().toString());
+        double accruedInterest = principal * ((double) interest) / 100;
+        return String.format("%.2f", accruedInterest);
+    }
+
+    @Override
+    public boolean checkUpdateDebt(Date currentDate) {
+        Calendar cal = Calendar.getInstance();
+        cal.setTime(lastAccruedDate);
+        int monthLastAccrued = cal.get(Calendar.MONTH);
+        cal.setTime(currentDate);
+        int currentMonth = cal.get(Calendar.MONTH);
+        // if currentMonth is 1 month ahead of monthLastAccrued
+        if ((currentMonth - monthLastAccrued == 1) || (currentMonth - monthLastAccrued == -11)) {
```

Comment:
what if the user has not opened the app for months?

Reply:
Good qn. haha i shall resolve it later

Reply:
nice wp

Reply:
Since its simple interest, we can use Java Date API to calculate the number of months between two dates and evaluate the interest.

Reply:
But i'm fine with the current state of it too, we can assume program is used everyday.